src/containers/execution-environment/registry-list.tsx

```diff
@@ -458,7 +462,8 @@ class ExecutionEnvironmentRegistryList extends React.Component<
       .catch(() =>
         this.addAlert(t`Failed to delete remote registry ${name}`, 'danger'),
       )
-      .then(() =>
+      .then(
+        () => this.queryRegistries(),
         this.setState({ showDeleteModal: false, remoteToEdit: null }),
```

Comment:
this change causes the modal to disappear right away, instead of waiting for the promise (because it's now evaulated right away, instead of when the callback gets called)

```suggestion
        () => { this.queryRegistries(); this.setState({ showDeleteModal: false, remoteToEdit: null }); }
```

Reply:
Hello:)
I tried this change for removing the cy.wait(10000), and unfortunately cypress still did not pass. I see this error:
`Timed out retrying after 4050ms: cy.click() failed because this element is detached from the DOM.

<button aria-label="Actions" id="pf-dropdown-toggle-id-13" class="pf-c-dropdown__toggle pf-m-plain" type="button" aria-expanded="false" aria-haspopup="true">...</button>

Cypress requires elements be attached in the DOM to interact with them.

The previous command that ran was:

  > cy.get()

This DOM element likely became detached somewhere between the previous and current command.

Common situations why this happens:
  - Your JS framework re-rendered asynchronously
  - Your app code reacted to an event firing and removed the element

You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands.Learn more
cypress/support/commands.js:626:9
  624 |           element.name +
  625 |           '"] button[aria-label="Actions"]',
> 626 |       ).click();
      |         ^
  627 |       cy.contains('a', 'Delete').click();
  628 |       cy.contains('button', 'Delete').click();
  629 |     });`